mqtt_client.c

```diff
@@ -990,6 +992,30 @@ static esp_err_t deliver_publish(esp_mqtt_client_handle_t client)
     return ESP_OK;
 }
 
+static esp_err_t deliver_suback(esp_mqtt_client_handle_t client)
+{
+    uint8_t *msg_buf = client->mqtt_state.in_buffer;
+    size_t msg_total_len = client->mqtt_state.message_length;
+    // SUBACK payload length = total length - (fixed header (2 bytes) + variable header (2 bytes))
```

Comment:
I'd suggest moving this to `mqtt_msg.h` file to isolate this _layer_ from an actual message formatting.

Reply:
much better indeed